app/questionnaire/questionnaire_store_updater.py

```diff
@@ -330,10 +378,14 @@ def update_progress_for_dependant_sections(self) -> None:
                 section_id != self._current_location.section_id
                 or list_item_id != self._current_location.list_item_id
             ):
-                self.update_section_status(
-                    is_complete=False,
-                    section_id=section_id,
-                    list_item_id=list_item_id,
+                # Remove any duplicate session dependents if they exist before setting
+                # the dependent section `is_complete` status to False
```

Comment:
Sorry if I wasn't clear. I mean, why we need to do this instead of what it does. The code is self explanatory, we know what it is doing but not why.